src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1987,6 +1998,23 @@ GenerateTaskMoveDependencyList(int64 colocationId,
 		hash_search(dependsList, &shardMoveDependencyInfo->taskId, HASH_ENTER, NULL);
 	}
 
+	/*
+	 * Check if there exists moves scheduled earlier whose source node
+	 * overlaps with the current move's target node.
```

Comment:
This deserves a more expansive comment on why this is considered a hard dependency. Because I don't think it's obvious.